### PR TITLE
EA-209 fix date comparison in getMothersAndChildren() to determine wh…

### PR DESCRIPTION
**Pull request by chibongho**
…ether child is born during mother's visit

See: https://openmrs.atlassian.net/browse/EA-209

### CODE REVIEW

File: api/src/main/resources/hql/mother_child.hql [hunk 1]
```diff
@@ -12,7 +12,6 @@ where
...
   and (:childRequiredToHaveActiveVisit = false or (select count(childVisit) from Visit as childVisit where childVisit.patient = child and childVisit.stopDatetime is null and childVisit.voided = false) > 0)
   and (:childRequiredToBeBornDuringMothersActiveVisit = false or (select count(motherVisit) from Visit as motherVisit where motherVisit.patient = mother and motherVisit.stopDatetime is null and motherVisit.voided = false
-                                                                                                                   and year(child.birthdate) >= year(motherVisit.startDatetime)
-                                                                                                                   and month(child.birthdate) >= month(motherVisit.startDatetime)
-                                                                                                                   and day(child.birthdate) >= day(motherVisit.startDatetime)) > 0)
+                                                                                                                   and (year(child.birthdate)||'-'||month(child.birthdate)||'-'||day(child.birthdate)) >=
```
Comment (chibongho) [inline api/src/main/resources/hql/mother_child.hql:15]: I don't like this manual concatenation, but I can't find a `DATE()` function in hql like in mysql.

**Review by mseaton — COMMENTED** [review]
Ick, ugly, but it was also ugly before.

**Comment by chibongho** [comment]
A bit moot, but looks like later version of hql has something saner:

    extract(date from child.birthdate)

https://docs.jboss.org/hibernate/orm/6.6/userguide/html_single/Hibernate_User_Guide.html

**Comment by mogoodrich** [comment]
Oh, thanks @chibongho , yeah, whoops, bug on my part, nice catch.  The string concanetation is ugly, but if we are assured that the greater than test will always work correctly, I guess we are okay.

The other option would be to be more explicit about the day/month/year comparisions, we could do this, right (admittedly "wordy", and I probably have my parathesis wrong, hard to match up outside of an editor)?

```
and (year(child.birthdate) >= year(motherVisit.startDatetime) or 

(year(child.birthdate) == year(motherVisit.startDatetime) and month(child.birthdate) >= month(motherVisit.startDatetime)) or

(year(child.birthdate) == year(motherVisit.startDatetime) and month(child.birthdate) == month(motherVisit.startDatetime)) and day(child.birthdate) >= day(motherVisit.startDatetime)) 
```

**Pull request merged**